### PR TITLE
Add support for deserializing arrays

### PR DESCRIPTION
**Pull request by jtbandes**
Adds support for fixed-length arrays.

Question for James: are there any other edge cases you can think of that should be tested?

### CODE REVIEW

File: src/test/ArraysTable.fbs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+struct Point3b {
```
Comment (jkuszmaul) [inline src/test/ArraysTable.fbs:1]: Good thing to do might be to set up some structs that could have weird alignment. E.g.

```
struct HasGaps {
  small:ubyte;
  // If I understand things right, there should be a 7 byte gap for padding here.
  big:[int64:3];
}
```

File: src/test/ArraysTable.fbs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+struct Mat3x3l {
+  cols:[Point3l:3];
+}
+struct Mat3x3f {
+  cols:[Point3f:3];
+}
+struct Mat3x3d {
+  cols:[Point3d:3];
+}
+
```
Comment (jkuszmaul) [inline src/test/ArraysTable.fbs:38]: * Array of enums

File: src/reflection.ts [hunk 3]
```diff
@@ -617,4 +629,61 @@ export class Parser {
       return result;
     };
   }
+
```
Comment (jkuszmaul) [inline src/reflection.ts:632]: TODO to self: haven't reviewed the actual logic yet.

File: src/test/ArraysTable.fbs (hunk 2 above)
Comment (jtbandes) [reply src/test/ArraysTable.fbs:38]: 😮 didn't realize that was allowed. I guess enums are "just" a scalar type under the hood.

**Comment by jtbandes** [comment]
I added some padding fields and verified that this does indeed result in some padding (see below). I'm not 100% sure this covers all the padding-related edge cases we want to test?

image

File: src/test/ArraysTable.fbs (hunk 2 above)
Comment (jtbandes) [reply src/test/ArraysTable.fbs:40]: Sadly I was unable to add this to the test because flatc crashes trying to generate the TS code. I'll probably file a ticket upstream.

**Comment by jkuszmaul** [comment]
> I added some padding fields and verified that this does indeed result in some padding (see below). I'm not 100% sure this covers all the padding-related edge cases we want to test?

This is what I was asking for; I don't know that it covers every possible weird padding/alignment thing, but mostly this should confirm that the new code is handling padding correctly so long as `flatc` is generating the schemas correctly.

File: src/test/ArraysTable.fbs (hunk 2 above)
Comment (jtbandes) [reply src/test/ArraysTable.fbs:40]: https://github.com/google/flatbuffers/issues/8311

**Review by jkuszmaul — APPROVED** [review]
